nipype/pipeline/plugins/multiproc.py

```diff
@@ -149,30 +129,19 @@ def __init__(self, plugin_args=None):
                                                        True)
 
         # Instantiate different thread pools for non-daemon processes
-        logger.debug('[MultiProc] Starting in "%sdaemon" mode (n_procs=%d, '
-                     'mem_gb=%0.2f, cwd=%s)', 'non' * int(non_daemon),
+        logger.debug('[MultiProc] Starting (n_procs=%d, '
+                     'mem_gb=%0.2f, cwd=%s)',
                      self.processors, self.memory_gb, self._cwd)
 
-        NipypePool = NonDaemonPool if non_daemon else Pool
-        try:
-            self.pool = NipypePool(
-                processes=self.processors,
-                maxtasksperchild=maxtasks,
```

Comment:
Apparently, there is no way to define a maxtasksperchild with concurrents.futures (https://stackoverflow.com/a/42229574). We should maybe consider loky.

Max tasks per child is important for FMRIPREP, as memory gets cleared up every time a worker is shut down and spun up.

Reply:
So if `concurrent.futures` doesn't have the features we need, perhaps it makes more sense to try to push them to fix the bug in `multiprocessing.Pool`?